### ISSUE TITLE
Feature Request - Exporting Favorites (database connections) to include passwords

### ISSUE THREAD

**Issue by ucsdmath** [sequelpro/sequelpro#2481]
When exporting a FAVORITES (under Quick Connect - Side bar) it would be great if it could include the password to re-import later on a clean install.

Keep up the great work!

**Comment by abhibeckert** [comment]
I agree there should be an option (disabled by default) to include passwords.

In the mean time, you cans imply search for "Sequel Pro" in Keychain Access to find all the passwords. Create a new empty keychain, copy/paste the entries into the new one, and you've got all your passwords in a file.

**Comment by abhibeckert** [comment]
Some more details after doing a bit of research:

In the export panel we should have a checkbox "include password". When selected, this will lock the keychain and then read the password from keychain (ensuring the user is required to enter their keychain password).

**Comment by quinn** [comment]
This would be a great feature to have. I would be happy to take a stab at this if someone could point me in the right direction.

**Comment by abhibeckert** [comment]
@quinn you'll need to add a checkbox enabling the feature to the accessory view of the export alert (see `ConnectionView.xib` and the `[SPConnectionController exportFavourites:]` method).

If it's checked, you'll need to lock the keychain (this may require using `NSTask` to execute `/usr/bin/security` with one argument `@"lock"`) so that your subsequent call to `[SPKeychain getPasswordForName:account:]` will prompt the user to unlock the keychain. Then simply include the passwords in the export data (created in `[SPTreeNode dictionaryRepresentation]`).

Obviously the import will also need to read the passwords, and use `SPKeychain` to add them.

**Comment by cobyrne09** [comment]
any luck with this @quinn?

**Comment by quinn** [comment]
@cobyrne09 no, i didn't make any real progress on it

**Comment by kaecyra** [comment]
My Sequel Pro list is something like 100 items. They each have their own password. I kind of want to just cry :(

**Comment by ucsdmath** [comment]
@kaecyra Wow! you are busy.  A full reinstall would take some time.  Even if you have the same server, but different databases, passwords, usernames, all configured-- it would still take a little time to setup again.   I have multiple machines I use (home, work, lab, etc.) and it is hard to keep them synced (or identical).  Always the challenge.

There is some need for this going forward.  Someone will find a way.  Reloading these configurations would be very nice and helpful.  So, break out that text file and cut-n-paste.  It is good practice!  Joking of course.

**Comment by dmoagx** [comment]
@kaecyra If your goal is to just transfer everything to a new Mac, you can simply copy the `~/Library/Application Support/Sequel Pro` folder to your new Mac, then go to Keychain Access, search for Sequel Pro, select all items, copy them, create a new Keychain and paste them into it (OS X may ask you for your Mac password for every single item, though). Copy the keychain file to your new Mac and open it with Keychain Access and then copy them back to the Login keychain.

**Comment by keevvinc** [comment]
@dmoagx are you sure by this? my `~/Library/Application Support/Sequel Pro`contains folders **bundles ; data ; themes** , but it **did not** recover my passwords

**Comment by dmoagx** [comment]
The AppSupport folder doesn't contain the passwords. That's why I explained the additional steps above.

**Comment by ucsdmath** [comment]
@dmoagx Thank you for elaborating on the Keychain files.  A very helpful method!
